rotkehlchen/chain/ethereum/interfaces/balances.py

```diff
@@ -34,19 +41,100 @@ def __init__(
             self,
             database: 'DBHandler',
             evm_inquirer: 'EvmNodeInquirer',
+            chain: 'SupportedBlockchain',
             counterparty: PROTOCOLS_WITH_BALANCES,
     ):
         self.counterparty = counterparty
         self.event_db = DBHistoryEvents(database)
         self.evm_inquirer = evm_inquirer
+        self.chain = chain
+
+    @abc.abstractmethod
+    def get_gauge_deposit_events(self) -> set[tuple[HistoryEventType, HistoryEventSubType]]:
+        """
+        Common method for all the classes implementing this interface. It returns the gauge
+        deposit events. Staking to a gauge is done with a gauge deposit transaction which
+        is decoded to a set of events. One of these events represents the deposit action.
+        This function should return the type and subtype of the event that represents this
+        deposit action.
+        """
 
     @abc.abstractmethod
-    def query_balances(self) -> dict[ChecksumEvmAddress, dict['EvmToken', 'Balance']]:
+    def get_gauge_address(self, event: 'EvmEvent') -> Optional[ChecksumEvmAddress]:
+        """
+        Common method for all the classes implementing this interface. It returns the gauge
+        address from the event that represents the gauge deposit action or None.
```

Comment:
The logic in this class is not only for gauges but for other contracts that allow to lock tokens and don't return a liquid version of the token itself, for example yearn stacked eth.

I would create a subclass for curve like protocols and not impose the subclasses of this interface to add a get_gauge_address method